core/trino-main/src/test/java/io/trino/execution/TestSqlTaskManagerRaceWithCatalogPrune.java

```diff
@@ -0,0 +1,255 @@
+import io.trino.connector.ConnectorServicesProvider;
+import io.trino.connector.MockConnectorFactory;
+import io.trino.connector.TestingLocalCatalogPruneTask;
+import io.trino.connector.WorkerDynamicCatalogManager;
+import io.trino.exchange.ExchangeManagerRegistry;
+import io.trino.execution.buffer.PipelinedOutputBuffers;
+import io.trino.execution.executor.dedicated.ThreadPerDriverTaskExecutor;
+import io.trino.execution.executor.scheduler.FairScheduler;
+import io.trino.memory.LocalMemoryManager;
+import io.trino.memory.NodeMemoryConfig;
+import io.trino.metadata.WorkerLanguageFunctionProvider;
+import io.trino.spi.connector.CatalogHandle;
+import io.trino.spi.connector.Connector;
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.spiller.LocalSpillManager;
+import io.trino.spiller.NodeSpillConfig;
+import io.trino.sql.planner.PlanFragment;
+import io.trino.testing.TestingConnectorContext;
+import io.trino.testing.TestingSession;
+import io.trino.transaction.NoOpTransactionManager;
+import io.trino.transaction.TransactionInfo;
+import io.trino.version.EmbedVersion;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+import static io.airlift.tracing.Tracing.noopTracer;
+import static io.trino.execution.BaseTestSqlTaskManager.OUT;
+import static io.trino.execution.TaskTestUtils.PLAN_FRAGMENT;
+import static io.trino.execution.TaskTestUtils.TABLE_SCAN_NODE_ID;
+import static io.trino.execution.TaskTestUtils.createTestSplitMonitor;
+import static io.trino.execution.TaskTestUtils.createTestingPlanner;
+import static io.trino.execution.buffer.PipelinedOutputBuffers.BufferType.PARTITIONED;
+import static io.trino.metadata.CatalogManager.NO_CATALOGS;
+import static io.trino.spi.connector.CatalogHandle.createRootCatalogHandle;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.version.EmbedVersion.testingVersionEmbedder;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.SAME_THREAD;
+
+@TestInstance(PER_CLASS)
+@Execution(SAME_THREAD)
+public class TestSqlTaskManagerRaceWithCatalogPrune
+{
+    private static final int NUM_TASKS = 40000;
+    private static final ConnectorServicesProvider NOOP_CONNECTOR_SERVICES_PROVIDER = new ConnectorServicesProvider()
+    {
+        @Override
+        public void loadInitialCatalogs()
+        {}
```

Comment:
For all of these you can put the empty curly brackets on the same like as the method, which will squish up the code a lot:
```suggestion
        public void loadInitialCatalogs() {}
```